clamav/util.py

```diff
@@ -70,32 +71,47 @@ def scan_stream(fileobj):
     return status, signature_or_none
 
 
-def scan_container_image(image_reference: str):
-    logger.debug(f'scanning container image {image_reference}')
+def yield_files_from_container_image(
+    container_image_reference: str,
+) -> typing.Tuple[typing.IO, str]:
+    '''Yields tuples of file and path from the container image
```

Comment:
please do not just re-iterate the keywords used in the function. this is not helping in comparison to just reading the source code.

instead, please document the abstract semantics.

which are:
- as input, a container image reference is accepted
- the container image is retrieved
- the contained layers are extracted in the order they are contained
- for each layer, two-tuples of contained file paths + file-like objects are returned (as a generator / iterable)

side note: it would also be acceptable for me to not add documentation. but if we do, i consider it very important for the documentation to be correct, precise, and actually add value compared to if the documentation were omitted.